packages/workflow/src/features/registration/fhir/fhir-bundle-modifier.ts

```diff
@@ -177,8 +182,48 @@ export async function invokeRegistrationValidation(
         ...headers
       }
     })
+    return { bundle }
   } catch (err) {
-    throw new Error(`Unable to send registration for validation: ${err}`)
+    const taskResource = getTaskResource(bundle)
+    const practitioner = await getLoggedInPractitionerResource(token)
+
+    if (
+      !taskResource ||
+      !taskResource.businessStatus ||
+      !taskResource.businessStatus.coding ||
+      !taskResource.businessStatus.coding[0] ||
+      !taskResource.businessStatus.coding[0].code
+    ) {
+      throw new Error('taskResource has no businessStatus code')
+    }
+    taskResource.businessStatus.coding[0].code = RegStatus.REJECTED
+
+    const statusReason: fhir.CodeableConcept = {
+      text: BIRTH_REG_NUMBER_GENERATION_FAILED
+    }
+    taskResource.statusReason = statusReason
+    taskResource.lastModified = new Date().toISOString()
+
+    /* setting registration workflow status here */
+    await setupRegistrationWorkflow(
+      taskResource,
+      getTokenPayload(token),
+      RegStatus.REJECTED
+    )
+
+    /* setting lastRegLocation here */
+    await setupLastRegLocation(taskResource, practitioner)
+
+    /* setting lastRegUser here */
+    setupLastRegUser(taskResource, practitioner)
+
+    /* check if the status of any event draft is not published and setting configuration extension*/
+    await checkFormDraftStatusToAddTestExtension(taskResource, token)
+
+    await postToHearth(bundle)
+    await triggerEvent(Events.BIRTH_MARK_VOID, bundle, headers)
```

Comment:
We want to void any event if there is a problem in country config, e.g.:

```
 await triggerEvent(
      getVoidEvent(event),
      bundle,
      headers
    )
```

With this util function:

```
function getVoidEvent(event: EVENT_TYPE): Events {
  switch (event) {
    case EVENT_TYPE.MARRIAGE:
      return Events.MARRIAGE_MARK_VOID
    case EVENT_TYPE.BIRTH:
      return Events.BIRTH_MARK_VOID
    case EVENT_TYPE.DEATH:
      return Events.DEATH_MARK_VOID
    default:
      return Events.BIRTH_MARK_VOID
  }
}
```